pkgs/development/libraries/concurrentqueue/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "concurrentqueue";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "cameron314";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1wj5i2yvdapbmpjmn0pyk2c9v03194zckxa6n5g9havqlsby78gs";
+  };
+
+  # The makefile uses relative paths so setting makefile or makeFlags doesn't work.
+  preBuild = ''
+    cd build
+  '';
+
+  #_FORTIFY_SOURCE requires compiling with optimization (-O)
+  NIX_CFLAGS_COMPILE = "-O";
```

Comment:
Do you know if you can set CFLAGS instead? NIX_CFLAGS_COMPILE should be the last resort.